go/vt/sqlparser/ast.go

```diff
@@ -1373,24 +1383,30 @@ func (node *BinaryExpr) WalkSubtree(visit Visit) error {
 
 // UnaryExpr represents a unary value expression.
 type UnaryExpr struct {
-	Operator byte
+	Operator string
 	Expr     Expr
 }
 
 // UnaryExpr.Operator
 const (
-	UPlusStr  = '+'
-	UMinusStr = '-'
-	TildaStr  = '~'
+	UPlusStr  = "+"
+	UMinusStr = "-"
+	TildaStr  = "~"
+	BangStr   = "!"
+	BinaryStr = "binary"
 )
 
 // Format formats the node.
 func (node *UnaryExpr) Format(buf *TrackedBuffer) {
 	if _, unary := node.Expr.(*UnaryExpr); unary {
-		buf.Myprintf("%c %v", node.Operator, node.Expr)
+		buf.Myprintf("%s %v", node.Operator, node.Expr)
 		return
 	}
-	buf.Myprintf("%c%v", node.Operator, node.Expr)
+	if node.Operator == BinaryStr {
+		buf.Myprintf("%s %v", node.Operator, node.Expr)
```

Comment:
this is pretty nit-picky, but honestly i think the extra space might be worth it to simplify the code.  an extra space in the case of `binary  <unary> 'foo'` doesn't really do anything.